web/src/ChatBox.js

```diff
@@ -70,7 +70,13 @@ class ChatBox extends React.Component {
   };
 
   handleRegenerate = () => {
-    const lastUserMessage = this.props.messages.reverse().find(message => message.author !== "AI");
+    const messages = this.props.messages;
+    if (messages.length === 1 && messages[0].replyTo === "Welcome" && messages[0].author === "AI") {
+      this.props.sendMessage("", "", true);
+      return;
+    }
+
+    const lastUserMessage = messages.reverse().find(message => message.author !== "AI");
     this.props.sendMessage(lastUserMessage.text, "", true);
```

Comment:
Do you really use `lastUserMessage.text`? Why not just use `this.props.sendMessage("Casibase##RegenerateFlag", "", true);`?

Reply:
Yes, it is required. In ChatPage.js, sendMessage() function, it needs text to generate a new message struct.